js/client-battle.js

```diff
@@ -379,8 +379,8 @@
 					if (this.request.forceSwitch !== true) {
 						var faintedLength = _.filter(this.request.forceSwitch, function (fainted) {return fainted;}).length;
 						var freedomDegrees = faintedLength - _.filter(switchables.slice(this.battle.mySide.active.length), function (mon) {return !mon.fainted;}).length;
-						this.choice.freedomDegrees = Math.max(freedomDegrees, 0);
-						this.choice.canSwitch = faintedLength - this.choice.freedomDegrees;
+						this.choice.canSwitch = faintedLength - Math.max(freedomDegrees, 0);
+						this.choice.freedomDegrees = this.battle.gen >= 7 ? 0 : Math.max(freedomDegrees, 0);
```

Comment:
I'm confused. Shouldn't `freedomDegrees` be sent from the server?

Reply:
That makes more sense. Though are you fine with that key name?

> In statistics, the number of degrees of freedom is the number of values in the final calculation of a statistic that are free to vary

Reply:
I really dislike the key name.

Reply:
Maybe the flag should be `canChooseSwitchLocation`.